cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtilFactory.java

```diff
@@ -16,10 +16,34 @@
 
 package co.cask.cdap.data2.util.hbase;
 
+import co.cask.cdap.common.conf.CConfiguration;
+import com.google.inject.Inject;
+
 /**
  * Factory for HBase version-specific {@link HBaseTableUtil} instances.
  */
 public class HBaseTableUtilFactory extends HBaseVersionSpecificFactory<HBaseTableUtil> {
+
+  private final CConfiguration cConf;
+
+  @Inject
+  public HBaseTableUtilFactory(CConfiguration cConf) {
+    this.cConf = cConf;
+  }
+
+  @Override
+  protected HBaseTableUtil createInstance(String className) throws ClassNotFoundException {
+    HBaseTableUtil hBaseTableUtil = super.createInstance(className);
+    hBaseTableUtil.setCConf(cConf);
+    return hBaseTableUtil;
+  }
+
+  public static Class<? extends HBaseTableUtil> getHBaseTableUtilClass() {
```

Comment:
I did not want the code using this to think that the factory could be instantiated without a parameter (because in that case, it is missing a CConf, and so operations on it won't be valid).